### PR TITLE
Add dataloader_idx argument to callbacks

### PR DESCRIPTION
**Pull request by maciej-sypetkowski**
## Changelogs

- Add `dataloader_idx` argument to pytorch lightning callbacks to fix errors when using certain pytorch lightning versions.

---

_Checklist:_

- [ ] _Was this PR discussed in an issue? It is recommended to first discuss a new feature into a GitHub issue before opening a PR._
- [ ] _Add tests to cover the fixed bug(s) or the new introduced feature(s) (if appropriate)._
- [ ] _Update the API documentation is a new function is added, or an existing one is deleted._
- [ ] _Write concise and explanatory changelogs above._
- [ ] _If possible, assign one of the following labels to the PR: `feature`, `fix` or `test` (or ask a maintainer to do it for you)._

---

_discussion related to that PR_

### CODE REVIEW

**Comment by DomInvivo** [comment]
Which lightning version are you referring to?

**Comment by maciej-sypetkowski** [comment]
I'm on 2.0.6 (almost the newest) and I encountered it. That's the newest signature https://lightning.ai/docs/pytorch/stable/common/lightning_module.html#on-validation-batch-end .

**Review by DomInvivo — CHANGES_REQUESTED** [review]
Also change the entry points in `ipu_wrapper.py`, class `PredictorModuleIPU`
